leptonai/config.py

```diff
@@ -100,11 +100,37 @@
 DB_PATH = CACHE_DIR / "lepton.db"
 LOGS_DIR = CACHE_DIR / "logs"
 
+
+def _is_rocm() -> bool:
+    """
+    Detects if we are using rocm.
+
+    If the user specified LEPTON_BASE_IMAGE_FORCE_ROCM to any true values, then
+    _is_rocm() returns True. If it is not specified, we rely on cuda to see if
+    we are building rocm. This function is only intended to be used to determine
+    the base image type, and you should directly use torch.version.hip in your
+    user code.
+    """
+
+    if "LEPTON_BASE_IMAGE_FORCE_ROCM" in os.environ:
+        true_values = ("yes", "true", "t", "1", "y", "on", "aye", "yea")
+        return os.environ["LEPTON_BASE_IMAGE_FORCE_ROCM"].lower() in true_values
+    else:
+        # Now, importing the whole torch might be an overkill if the user is not
+        # using any torch capabilities, but most of lepton use cases would probably
+        # involve torch, and we will just use it.
+        import torch
```

Comment:
Most of the time `LEPTON_BASE_IMAGE_FORCE_ROCM` is not set, and this will cause every cli invocation run `import torch` and slow

Reply:
I think we should put both cuda and rocm base image tage in the generated photon, and platform chooses which base image to use at deployment time.